yellowbrick/features/rankd.py

```diff
@@ -240,27 +275,173 @@ def rank(self, X, algorithm=None):
                 "'{}' is unrecognized ranking method".format(algorithm)
             )
 
+        # Extract matrix from dataframe if necessary
+        if is_dataframe(X):
+            X = X.as_matrix()
+
         return self.ranking_methods[algorithm](X)
 
-    def draw(self, X, **kwargs):
+    def finalize(self, **kwargs):
+        """
+        Finalize executes any subclass-specific axes finalization steps.
+        The user calls poof and poof calls finalize.
+
+        Parameters
+        ----------
+        kwargs: dict
+            generic keyword arguments
+
+        """
+        # Set the title
+        self.set_title(
+            "{} Ranking of {} Features".format(
+                self.ranking_.title(), len(self.features_)
+            )
+        )
+
+
+##########################################################################
+## Rank 1D Feature Visualizer
+##########################################################################
+
+class Rank1D(RankDBase):
+    """
+    Rank1D computes a score for each feature in the data set with
+    a specific metric or algorithm (e.g. takes the shapiro) then returns
+    them ranked as a bar plot.
+
+    Parameters
+    ----------
+    ax : matplotlib Axes, default: None
+        The axis to plot the figure on. If None is passed in the current axes
+        will be used (or generated if required).
+
+    algorithm : one of {'shapiro', }, default: 'shapiro'
+        The ranking algorithm to use, default is 'Shapiro-Wilk.
+
+    features : list
+        a list of feature names to use
+        If a DataFrame is passed to fit and features is None, feature
+        names are selected as the columns of the DataFrame.
+
+    colormap : string or cmap, default: 'RdBu_r'
+        optional string or matplotlib cmap to colorize lines
+        Use either color to colorize the lines on a per class basis or
+        colormap to color them on a continuous scale.
+
+    kwargs : dict
+        Keyword arguments that are passed to the base class and may influence
+        the visualization as defined in other Visualizers.
+
+    Examples
+    --------
+
+    >>> visualizer = Rank2D()
+    >>> visualizer.fit(X, y)
+    >>> visualizer.transform(X)
+    >>> visualizer.poof()
+    """
+    ranking_methods = {
+        'shapiro': lambda X: np.array([shapiro(x)[0] for x in X.T]),
+    }
+
+    def __init__(self, ax=None, algorithm='shapiro', features=None,
+                 colormap='RdBu_r', **kwargs):
+        """
+        Initialize the class with the options required to rank and
+        order features as well as visualize the result.
+        """
+        super(Rank1D, self).__init__(ax=None, algorithm=algorithm, features=features,
+                         colormap=colormap, **kwargs)
+
+    def draw(self, R, **kwargs):
+        """
+        Draws the heatmap of the ranking matrix of variables.
+        """
+        self.ax.bar(np.arange(len(R)), R, color=self.colormap[0])
```

Comment:
This is very minor, but I'm worried that using the first indexed color of the colormap will produce variable results (and this can make testing very tricky, beyond simply annoying users a bit). Since you're only using a single color, I recommend simply using the value `'b'` here -- the discrete color cycle in matplotlib is `'bgrmyck'` and so `'b'` is always the first color we use for yb visualizations (giving them coordinated look and feel). Note that using this notation the `'b'` will be equivalent to the `set_palette()` color `'b'` and will therefore maintain the user's style preferences.